packages/eslint-plugin/src/rules/no-restricted-imports/index.js

```diff
@@ -0,0 +1,165 @@
+// @ts-check
+const { AST_NODE_TYPES } = require('@typescript-eslint/experimental-utils');
+const createRule = require('../../utils/createRule');
+
+/**
+ * @typedef {import("@typescript-eslint/types/dist/ts-estree").ImportClause} ImportClause
```

Comment:
would add some documentation to this type like `Lookup for insertion point for new imports when moving a restricted import to a preferred import`